### PR TITLE
feat(hourly): Add configuration to enable hourly jobs

### PR DESCRIPTION
**Pull request by mdschmitt**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

Not unless existing pillar values for clients have "hourly" set in their job configurations.  ...in which case, they already don't work straight-out-of-the-box.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Review by myii — CHANGES_REQUESTED** [review]
Thanks for the PR, @mdschmitt.  Some initial feedback for the time being.

After numerous discussions, essentially we _never_ use `salt['pillar.get']()` in our formulas.  Let me give some brief explanations:

1. `salt['pillar.get']()` is unnecessary since `map.jinja` already merges all of the elements under one key, which is [`logrotate`](https://github.com/saltstack-formulas/logrotate-formula/blob/da488f4bf75518d8018193792da0f7d510d82c43/logrotate/map.jinja#L45) for this formula.
1. There have been many reasons given why non-secret data shouldn't be stored under pillars (e.g. pillars are rendered on the master); so we're in the process of moving to `config.get` instead, which has already been adopted by this formula in [`map.jinja`](https://github.com/saltstack-formulas/logrotate-formula/blob/da488f4bf75518d8018193792da0f7d510d82c43/logrotate/map.jinja#L13).

Hence, using `salt['pillar.get']()` inline is unnecessary duplication, longer to write, more difficult to review (not a single point for all data being used by the formula), not to mention prevents end-users from providing their data from sources other than pillar (e.g. SDB).

You should find it relatively simple to access the data via the `logrotate` key.  Let me know if you face any issues.  Once you're done, please amend/rebase your commit(s) and force-push back here -- thanks.

File: logrotate/config.sls [hunk 1]
```diff
@@ -22,3 +32,33 @@ logrotate-directory:
     - mode: 755
     - makedirs: True
 
+{% if ns.hourly == True %}
```
Comment (myii) [inline logrotate/config.sls]:
```suggestion
{%- if ns.hourly %}
```

The above is sufficient.  Note also the use of Jinja whitespace control (`{%- ... %}`).  Again, there have been a number of discussions about this and we've found this structure to be effective in nearly all situations.  There are rare cases where you need to apply the post-whitespace control (`{% ... -%}`) but we're agreed that we should never use pre- and post-whitespace control (`{%- ... -%}`).

One of the main benefits of this is working through the debug output, showing how states have been rendered.  It is also very useful in templates.

**Comment by mdschmitt** [comment]
@myii , how's this look?

**Comment by mdschmitt** [comment]
No major rush, but any word on this?

**Comment by mdschmitt** [comment]
@myii , dropping another *bump* in here.  Any feedback on my revisions?

**Comment by mdschmitt** [comment]
Latest CI failure because ArchLinux properly uses systemd timers rather than cron.  Requires deeper code tweakage to properly provide for.  This PR on hold until I can get that to be better.

**Comment by myii** [comment]
@mdschmitt Thanks for the additions.  I've tested some changes in my own fork to ensure Arch Linux is working as well.  I'll push those here and merge this PR shortly.

One question remains: which reference did you use for the `/etc/logrotate.hourly.d/...` path?  I couldn't find any mention of it in the upstream repo:

* https://github.com/logrotate/logrotate/search?q=%22logrotate.hourly.d%22&type=code

**Review by myii — APPROVED** [review]

**Comment by mdschmitt** [comment]
Wow @myii , thank you so much for doing this.  You beat me to writing the kitchen test and I definitely would have ended up rewriting things to allow for Arch.  Folding in `cron-formula` as a dependency is much cleaner though.

As far as the naming for "logrotate.hourly.d", it was just a logical canonicalization of "logrotate.d" with the explicit addition of "hourly" since hourly isn't out-of-the-box functionality.  Inspiration taken from https://sleeplessbeastie.eu/2018/07/11/how-to-execute-logrotate-every-hour

**Comment by myii** [comment]
No problem -- thanks for the reference, @mdschmitt.

**Pull request merged**